src/Hamiltonians/ExtendedHubbardReal1D.jl

```diff
@@ -33,6 +33,14 @@ function starting_address(h::ExtendedHubbardReal1D)
     return getfield(h, :add)
 end
 
+# `ExtendedHubbardReal1D` conserves particle number. Thus we can lower the bound on the dimension
+# for the non-conserving `OccupationNumberFS`.
+function dimension(::ExtendedHubbardReal1D, a::OccupationNumberFS)
+    m = num_modes(a)
+    n = num_particles(a)
+    return dimension(BoseFS{n,m})
+end
+
```

Comment:
I don't like having to repeat this snippet for every Hamiltonian, but I guess there's no better way at the moment.